redbox-core/redbox/chains/components.py

```diff
@@ -16,22 +16,16 @@
 from redbox.retriever import AllElasticsearchRetriever, ParameterisedElasticsearchRetriever, MetadataRetriever
 from langchain_community.embeddings import BedrockEmbeddings
 from langchain.chat_models import init_chat_model
+from redbox.models.chain import ChatLLMBackend
 
 
 logger = logging.getLogger(__name__)
 load_dotenv()
 
 
-def get_chat_llm(model: str, tools: list[StructuredTool] | None = None):
-    if model.startswith("gpt-"):
-        model_provider = "azure_openai"
-    elif model.startswith("anthropic."):
-        model_provider = "bedrock"
-    else:
-        raise ValueError("%s not recognised", model)
-
-    logger.info("initialising model=%s model_provider=%s tools=%s", model, model_provider, tools)
-    chat_model = init_chat_model(model=model, model_provider=model_provider)
+def get_chat_llm(model: ChatLLMBackend, tools: list[StructuredTool] | None = None):
+    logger.info("initialising model=%s model_provider=%s tools=%s", model.name, model.provider, tools)
+    chat_model = init_chat_model(model=model.name, model_provider=model.provider)
```

Comment:
`init_chat_model` takes `configurable_fields`  [https://python.langchain.com/api_reference/langchain/chat_models/langchain.chat_models.base.init_chat_model.html#init-chat-model](parameter). Do you think we should support passing this in? It does warn of a security risk being able to change the base_url and redirect requests. The only initial use case I can think of is over writting the base url for `ollama` if you are running the service on apple silicon and the service in a docker container, as you have to route from the docker container to the host machine.

Reply:
this sounds reasonable. how about:

```python
chat_model = init_chat_model(model=model.name, model_provider=model.provider, configurable_fields=["base_url"])
```

to begin with and we can add more if we need them?

Reply:
[3177410](https://github.com/i-dot-ai/redbox/pull/1084/commits/317741033e571a91e24a01e2fe80bb24460d2de5)